adhocteam/Head-Start-TTADP
# CI: Add parallel run, test view and coverage report

**Description of change**
Follow along PR to https://github.com/adhocteam/Head-Start-TTADP/pull/9

Adds three features to Continuous Integration (CI)
 1. Runs three jobs in parallel: frontend (FE) and backend (BE) linting, FE testing, and BE testing. Right now this is making the CI pipeline take more time to run because persisting and reinstating copies of the build workspace takes time (< 1 minute).  I expect this added expense to be static. After we add more FE and BE tests, and the time to run either suite takes > 1 min, we'll start experiencing the benefits of the parallel testing structure.
![Screen Shot 2020-08-25 at 2 44 29 PM](https://user-images.githubusercontent.com/9893700/91220316-9a5c6d00-e6e1-11ea-8da2-5cfe7c921915.png)

2. Testing summary is available under the job "tests" view
![Screen Shot 2020-08-25 at 2 47 33 PM](https://user-images.githubusercontent.com/9893700/91220547-f1624200-e6e1-11ea-9e54-0d641ab5c72d.png)

3. Coverage reports are viewable/downloadable under the job "artifacts" view
![Screen Shot 2020-08-25 at 2 48 28 PM](https://user-images.githubusercontent.com/9893700/91220670-1d7dc300-e6e2-11ea-9778-054feff20e05.png)

**How to test**
Check out [latest circle ci build](https://app.circleci.com/pipelines/github/adhocteam/Head-Start-TTADP/84/workflows/e5f36c49-b1ed-447e-b77e-642e143fdec0) of this branch

**Issue(s)**
Part of https://github.com/HHS/Head-Start-TTADP/issues/12

**Checklist**

- [n/a] Meets issue criteria
- [n/a] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [n/a] Documentation updated
    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified
